backend/lib/endpoints/posts.js

```diff
@@ -73,13 +73,41 @@ router.post(
     req.body.authorId = req.user.id;
     req.body.postId = postId;
     new Comment(req.body).save().then((comment) => {
-      Post.findOneAndUpdate({ _id: postId }, { $push: { comments: comment } })
-        .then((post) => res.status(200).json(comment))
+      Post.findOneAndUpdate(
+        { _id: postId },
+        { $push: { comments: comment } },
+        { new: true },
+      )
+        .then((post) => res.status(200).json(post))
         .catch((err) => res.status(404).send(err));
     });
   },
 );
 
+/**
+ * @route POST api/posts/:postId/like
+ * @desc Like/Unlike a post once
+ * @access Protected
+ */
+router.post(
+  "/:postId/like",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    Post.findOne({ _id: req.params.postId }, (err, post) => {
+      if (err) res.send(err);
+      const userId = req.user.id;
+      if (!post.likes.includes(userId)) {
```

Comment:
yeah so this is probably good for first version
we could also add a table "likes" with { likeId, userId, postId }

Reply:
also small potential improvement but you could use only `indexOf`check probably and if / else without `includes` :)

Reply:
@robinv85 Thanks for the suggestions! I'll look into them when I get a chance. 👍

Reply:
So if you fetch a list of posts and you want to show the user if they have liked it or not, it can get very big and slow I think. Would be best to index both on like id, post id and user id in that case.